### ISSUE TITLE
`enabled={false}` on `<Editor/>` does not take effect if `json` prop is set to `<Frame/>`

### ISSUE THREAD

**Issue by aptarmy** [prevwong/craft.js#395]
**Describe the bug**
This issue is similar to #373, but a little bit different.

`enabled={false}` prop on `<Editor/>` is working fine if there is no prop like `json` or `data` set on `<Frame/>`.

But things started to break if preloaded data is set to `<Frame/>` through `json` or `data` prop, any immediet elements inside `<Element canvas/>` can be dragable regardless of `enabled` prop on `<Editor/>`

**Expected behavior**
There shouldn't be anything dragable if `enabled={false}` and whether preloaded data is set or not.

**My unsuccessful workarounds**
I tried using the following pieces of code but the issue still persists.
```
// force disabling editor on ComponentDidMount doesn't fix the issue
useEffect(()=>{
  actions.setOptions((options) => (options.enabled = false));
}, []);
```
or
```
// may be turning it on and off again can fix the issue but nothing fixed
useEffect(()=>{
  actions.setOptions((options) => (options.enabled = true));
  actions.setOptions((options) => (options.enabled = false));
}, []);
```
or
```
// force disabling dragable area doesn't fix the issue, immediet nodes can still be dragable
const { enabled } = useEditor(state => ({ enabled: state.options.enabled })); // when enabled is false
<Element canvas={enabled}>
```

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| craft.js         | 0.2.0-beta.3
| React            | 16.8.0
| TypeScript       | -
| Browser          | Chrome 99.0.4844.74 (Official Build) (64-bit)
| npm/Yarn         |  8.5.0
| Operating System | Ubuntu 18.04

Thank you for making this amazing tool anyway. I appreciate your time.

**Comment by aptarmy** [comment]
I finally find a dirty workaround.
For those who are facing the same issue, try this work around. My code is working fine now.

Do this when you want your editor to be in `disabled` state. The bug is from `json` or `data` prop from `Frame` element, so the workaround is to render React Element yourself using `json` data derrived from `query.serialize()`. And produce new `key` prop to `<Editor/>` element, it will refresh your editor with new instance and this bug will disaapear.

Use below code to render `json` data yourself

```
const editorResolver = { Card, Button, Container, ...yourResolver };
const renderPreloadedJson = jsonData => {
  jsonData = JSON.parse(jsonData);
  const renderElm = nodeData => {
    return React.createElement(editorResolver[nodeData.type.resolvedName], nodeData.props, ...nodeData.nodes.map(node => renderElm(jsonData[node])));
  };
  return renderElm(jsonData['ROOT']);
}
```

And when you want your editor to be in `enabled` state, you can set json string directly to `json` or `data` prop on `<Frame/>` element and generate new `key` for `<Editor/>` to refresh your editor with new instance.

**Comment by aptarmy** [comment]
Ohh the workaround from #373 solve the issue 

```
const { enabled, actions } = useEditor(state => ({ enabled: state.options.enabled }));
  useEffect(() => {
    setTimeout(() => {
      actions.setOptions(options => options.enabled = !enabled);
      actions.setOptions(options => options.enabled = enabled);
    });
  }, []);
```

**Comment by MarkLyck** [comment]
@aptarmy that doesn't "solve" the issue. That is a hacky workaround.

`enabled={false}` should disable dragging in the entire Editor.

**Comment by aptarmy** [comment]
@MarkLyck Yes it is, but it is better to only have one issue at a time so I close this issue, otherwise it will duplicate #373
Please subscribe the referenced issue for more updates, Thanks.

**Comment by prevwong** [comment]
Released in v0.2.0-beta.4 🎉

**Comment by aptarmy** [comment]
Thank you!
